x-pack/plugins/stack_alerts/public/alert_types/es_query/validation.ts

```diff
@@ -111,5 +75,66 @@ export const validateExpression = (alertParams: EsQueryAlertParams): ValidationR
       })
     );
   }
+
+  /**
+   * Skip esQuery and index params check if it is search source alert,
+   * since it should contain searchConfiguration instead of esQuery and index.
+   */
+  const isSearchSource = isSearchSourceAlert(alertParams);
+  if (isSearchSource) {
+    if (!alertParams.searchConfiguration) {
+      errors.searchConfiguration.push(
+        i18n.translate(
+          'xpack.stackAlerts.esQuery.ui.validation.error.requiredSearchConfiguration',
+          {
+            defaultMessage: 'Search source configuration is required.',
+          }
+        )
+      );
+    }
+    return validationResult;
+  }
+
+  if (!alertParams.index || alertParams.index.length === 0) {
+    errors.index.push(
+      i18n.translate('xpack.stackAlerts.esQuery.ui.validation.error.requiredIndexText', {
+        defaultMessage: 'Index is required.',
+      })
+    );
+  }
+
+  if (!alertParams.timeField) {
+    errors.timeField.push(
+      i18n.translate('xpack.stackAlerts.esQuery.ui.validation.error.requiredTimeFieldText', {
+        defaultMessage: 'Time field is required.',
+      })
+    );
+  }
+
+  if (!alertParams.esQuery) {
+    errors.esQuery.push(
+      i18n.translate('xpack.stackAlerts.esQuery.ui.validation.error.requiredQueryText', {
+        defaultMessage: 'Elasticsearch query is required.',
+      })
+    );
+  } else {
+    try {
+      const parsedQuery = JSON.parse(alertParams.esQuery);
+      if (!parsedQuery.query) {
+        errors.esQuery.push(
+          i18n.translate('xpack.stackAlerts.esQuery.ui.validation.error.requiredEsQueryText', {
+            defaultMessage: `Query field is required.`,
```

Comment:
I'm not sure where this shows up.  The other messages don't include the word "field".  So can this be:

> Query is required.

Reply:
I would suggest to keep current message, since it appears when request validation fails and actually applies to missing to query object field. WDYT @kertal ?

Reply:
There are other messages here like `Time field is required.`, so I think it's fine this way